File: 2299-merge-nodes-in-between-zeros/2299-merge-nodes-in-between-zeros.php
class Solution {
  function mergeNodes($head) {
    $node = $head->next;
    $head = new ListNode();
    $tail = $head;
    $sum = 0;
    while ($node !== null) {
      if ($node->val !== 0) {
        $sum += $node->val;
      } else {
        $tail->next = new ListNode($sum);
        $tail = $tail->next;
        $sum = 0;
      }
      $node = $node->next;
    }
    return $head->next;
  }
}

File: 2299-merge-nodes-in-between-zeros/2299-merge-nodes-in-between-zeros.py
class Solution:
  def mergeNodes(self, node: Optional[ListNode]) -> Optional[ListNode]:
    head = ListNode(0)
    tail = head
    rs = 0
    node = node.next
    while node:
      rs += node.val
      if not node.val:
        tail.next = ListNode(rs)
        tail = tail.next
        rs = 0
      node = node.next
    return head.next

        